x-pack/plugins/ml/public/services/new_job_capabilities_service.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexPattern } from 'ui/index_patterns';
+import { Field, Aggregation, AggId, FieldId, NewJobCaps } from '../../common/types/fields';
+import { ml } from './ml_api_service';
+
+// the type property is missing from the official IndexPattern interface
+interface IndexPatternWithType extends IndexPattern {
```

Comment:
Could this be of use in other parts of ML? If so, I wonder if it could under the `common/types` folder?